### ISSUE TITLE
QuickLaTeX cannot compile formula Error

### ISSUE THREAD

**Issue by johnpeterm** [BCcampus/pressbooks-openstax-import#73]
### Description 

On some section of the formula, the following error occurs
*** QuickLaTeX cannot compile formula:
\begin{array}{}\\ \\ \phantom{\rule{1.2em}{0ex}}7.56\phantom{\rule{0.2em}{0ex}}\text{kg}\hfill \\ -6.052\phantom{\rule{0.2em}{0ex}}\text{kg}\hfill \\ \\ \\ \phantom{\rule{0.2em}{0ex}}\frac{\phantom{\rule{0.2em}{0ex}}+13.7\phantom{\rule{0.2em}{0ex}}\text{kg}}{15.208\phantom{\rule{0.2em}{0ex}}\text{kg}}=15.2\phantom{\rule{0.2em}{0ex}}\text{kg}\text{.}\hfill \end{array}

*** Error message:
Missing # inserted in alignment preamble.
leading text: $\begin{array}{}
Missing $ inserted.
leading text: ...15.208\phantom{\rule{0.2em}{0ex}}\text{kg}}
Extra }, or forgotten $.
leading text: ...15.208\phantom{\rule{0.2em}{0ex}}\text{kg}}
Missing } inserted.
leading text: ...m}{0ex}}\text{kg}\text{.}\hfill \end{array}
Extra }, or forgotten $.
leading text: ...m}{0ex}}\text{kg}\text{.}\hfill \end{array}
Missing } inserted.
leading text: ...m}{0ex}}\text{kg}\text{.}\hfill \end{array}
Extra }, or forgotten $.
leading text: ...m}{0ex}}\text{kg}\text{.}\hfill \end{array}
Missing } inserted.
leading text: ...m}{0ex}}\text{kg}\text{.}\hfill \end{array}
Extra }, or forgotten $.
leading text: ...m}{0ex}}\text{kg}\text{.}\hfill \end{array}
Missing } inserted.

**Expected behaviour** 

![12](https://user-images.githubusercontent.com/20897821/42831090-0e9c0a8a-8a0b-11e8-849a-5377f114281a.png)

**Actual behaviour** 

[What actually happened]

### Steps to reproduce the problem

[First Step, Second Step, etc]

### System Information

- Openstax Import for Pressbooks: [Enter POI Version here] 
Pressbooks: [Version 5.4.0]
WordPress: [Version 4.9.7]

**Comment by bdolor** [comment]
I can reproduce this - the conversion process is from `mathml` (in the original book) to `LaTeX` via `xsl`: https://github.com/BCcampus/pressbooks-openstax-import/tree/dev/inc/import/openstax/xsl

The LaTeX is then rendered via QuickLaTeX image rendering service. Unfortunately equation conversion is not 100% all the time, especially in books with many equations. That said, we've been happy with the results, albeit imperfect.

**Comment by johnpeterm** [comment]
Despite imperfection, it is huge time saver.

**Comment by bdolor** [comment]
@johnpeterm - thanks for mentioning that. We agree it's better than it was when it was a purely manual process. Converting an openstax book with equations would take months. That said, there are plans to incrementally improve the performance of this importer, so I'll leave this ticket open for now. Though I'm not sure how it will affect equation transformation, we hope to see some improvements with https://github.com/BCcampus/pressbooks-openstax-import/issues/71 Thanks for reporting the issue.

**Comment by otterscotter** [comment]
None of the formulas in this conversion are displaying properly after the import. Any ideas?
Code in the editor:
The magnitude of the charge is independent of the type. Phrased another way, the smallest possible positive charge (to four significant figures) is $+1.602\times10^{-19}~\mathrm{C}$, and the smallest possible negative charge is $-1.602\times10^{-19}~\mathrm{C}$; these values are exactly equal. This is simply how the laws of physics in our universe turned out.
As displayed in the book:
The magnitude of the charge is independent of the type. Phrased another way, the smallest possible positive charge (to four significant figures) is $+1.602\times10^{-19}~\mathrm{C}$, and the smallest possible negative charge is $-1.602\times10^{-19}~\mathrm{C}$; these values are exactly equal. This is simply how the laws of physics in our universe turned out.
https://pbooks.scottrobison.net/physics155/chapter/electric-charge

**Comment by otterscotter** [comment]
Yes, version 3.8.5. Might there be any configuration changes needed? I'm a
complete LaTeX noob.

[image: Sketch (54).png]


*Scott Robison, Ph.D.*
*(pronouns: he, him, his)*
Associate Director, Digital Learning and Design
Portland State University
Portland, OR 97201
503-725-9118
@otterscotter

**Comment by otterscotter** [comment]
Sorry, doesn't look like the screen shot went through...
![sketch 54](https://user-images.githubusercontent.com/37352212/46319432-e7d00e00-c58e-11e8-94e5-2069e2f1b395.png)

**Comment by bdolor** [comment]
If the WP QuickLaTeX plugin was activated prior to and during the import process, then there is no other trickery required. If this is something you can confirm this is an issue with the plugin then I encourage you to create a new issue with an eye to giving me enough detail to re-create the issue. If it is a general support request though, I'm afraid I don't have the capacity to be much help.

**Comment by otterscotter** [comment]
Ha. I just imported Calculus Volume 1 and at first glance, formulas display properly! I'll try re-importing problematic ones again. Thanks!
